tools/undelete-block-gcs/main.go

```diff
@@ -1,130 +0,0 @@
-// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
I don't think we should delete this tool just yet. I think it's still useful, and much simpler to work with.

Reply:
> I don't think we should delete this tool just yet. I think it's still useful, and much simpler to work with.

If you apply equal standards to each tool then this doesn't make sense. Not dealing with complexity doesn't make it go away. The GCS tool has a number of drawbacks that this tool does not:

-  Despite the name being `undelete-block-gcs` there is nothing tying the input to be on block prefixes. This is not a virtue since its entirely on the user to prepare the undocumented input format.
-  It does not handle multiple versions of an object gracefully. If there is a noncurrent version that exists, it restores it. Even if a current version exists for the same object it does this. When there are multiple noncurrent versions for the same object it does this repeatedly, overwriting itself and generating even more noncurrent versions. In particular this is a barrier to reruns with the same input if an error is encountered.
- It will restore any object (besides a delete marker) with a noncurrent version within the block prefix, even when restoring the block fully is impossible. If a chunk object had no noncurrent versions that restored block is now corrupt. If a block had been backfilled it would restore the temporary objects backfill adds.
- Deleting delete markers (both local and global) is unhandled. It assumes the block is fully deleted.

Despite the above, the GCS tool did its job when it needed to very successfully. What I'm trying to do here is iterate and solve these problems to make it more accessible. It's more lines of code, sure, but there's a reason for it and I want us to align on that.